### PR TITLE
added CSV import/export of server blocklist

### PR DESCRIPTION
**Pull request by tobiasd**
This PR adds two new subcommands to the console command `serverblock` that can be used by the node admin to import and export the nodes server blocklist from/to a CSV file.

```
bin/console serverblock import test.csv
bin/console serverblock export test.csv
```

Part of #8881

### CODE REVIEW

**Comment by annando** [comment]
Please update the test here as well: https://github.com/friendica/friendica/blob/develop/tests/src/Console/ServerBlockConsoleTest.php#L328

**Comment by tobiasd** [comment]
What is the test supposed to do? Check that the `--help` text is the same as it was copied over somewhere?

**Comment by MrPetovan** [comment]
Yes.

**Comment by tobiasd** [comment]
ok, lets see if I can copy&paste the correct parts of the text or if it is already too hot ;-)

**Comment by annando** [comment]
Have you tested what happens when a reason contains the separator sign for the CSV?

**Comment by MrPetovan** [comment]
> Have you tested what happens when a reason contains the separator sign for the CSV?

In this case the field should be enclosed with quotes.

File: src/Console/ServerBlock.php [hunk 1]
```diff
@@ -48,14 +48,18 @@ protected function getHelp()
 		$help = <<<HELP
 console serverblock - Manage blocked server domain patterns
 Usage
-	bin/console serverblock [-h|--help|-?] [-v]
-	bin/console serverblock add <pattern> <reason> [-h|--help|-?] [-v]
-	bin/console serverblock remove <pattern> [-h|--help|-?] [-v]
+    bin/console serverblock [-h|--help|-?] [-v]
```
Comment (MrPetovan) [inline src/Console/ServerBlock.php:51]: Why did you change the indentation style?

File: src/Console/ServerBlock.php [hunk 2]
```diff
@@ -87,17 +91,77 @@ protected function doExecute()
...
+	 * file, e.g. created with the export function.
 	 *
 	 * @param IConfig $config
 	 */
+	private function importBlockedServers(IConfig $config)
+	{
+		$filename = $this->getArgument(1);
+		$currBlockList = $config->get('system', 'blocklist', []);
+		$newBlockList = [];
+		if (($fp = fopen($filename, 'r')) !== FALSE) {
```
Comment (MrPetovan) [inline src/Console/ServerBlock.php:130]: Please add an error message if the file can't be open. Additionally, we use lowercase `false`.
```suggestion
		if (($fp = fopen($filename, 'r')) !== false) {
```

File: src/Console/ServerBlock.php [hunk 3]
```diff
@@ -87,17 +91,77 @@ protected function doExecute()
...
+			while (($data = fgetcsv($fp, 1000, ',')) !== FALSE) {
+				$domain = $data[0];
+				if (count($data)  == 0) {
```
Comment (MrPetovan) [inline src/Console/ServerBlock.php:133]:
```suggestion
				if (count($data) == 0) {
```

File: src/Console/ServerBlock.php [hunk 4]
```diff
@@ -87,17 +91,77 @@ protected function doExecute()
...
+				$data = [
+					'domain' => $domain,
+					'reason' => $reason
+				];
+				if (!in_array($data, $newBlockList))
+					$newBlockList[] = $data;
+			}
+			foreach ($currBlockList as $blocked) {
+				if (!in_array($blocked, $newBlockList))
+					$newBlockList[] = $blocked;
```
Comment (MrPetovan) [inline src/Console/ServerBlock.php:147]: Code standards: Please wrap conditionals with curly braces.
```suggestion
				if (!in_array($blocked, $newBlockList)) {
					$newBlockList[] = $blocked;
				}
```

File: src/Console/ServerBlock.php [hunk 5]
```diff
@@ -87,17 +91,77 @@ protected function doExecute()
...
+					$reason = self::DEFAULT_REASON;
+				} else {
+					$reason = $data[1];
+				}
+				$data = [
+					'domain' => $domain,
+					'reason' => $reason
+				];
+				if (!in_array($data, $newBlockList))
+					$newBlockList[] = $data;
```
Comment (MrPetovan) [inline src/Console/ServerBlock.php:143]: Code standards: Please wrap conditionals with curly braces.
```suggestion
				if (!in_array($data, $newBlockList)) {
					$newBlockList[] = $data;
				}
```

File: src/Console/ServerBlock.php [hunk 6]
```diff
@@ -87,17 +91,77 @@ protected function doExecute()
...
 	 *
 	 * @param IConfig $config
 	 */
+	private function importBlockedServers(IConfig $config)
+	{
+		$filename = $this->getArgument(1);
+		$currBlockList = $config->get('system', 'blocklist', []);
+		$newBlockList = [];
+		if (($fp = fopen($filename, 'r')) !== FALSE) {
+			while (($data = fgetcsv($fp, 1000, ',')) !== FALSE) {
```
Comment (MrPetovan) [inline src/Console/ServerBlock.php:131]:
```suggestion
			while (($data = fgetcsv($fp, 1000, ',')) !== false) {
```

File: src/Console/ServerBlock.php [hunk 7]
```diff
@@ -87,17 +91,77 @@ protected function doExecute()
...
+	 * Exports the list of blocked domains including the reason for the
+	 * block to a CSV file.
+	 *
+	 * @param IConfig $config
+	 */
+	private function exportBlockedServers(IConfig $config)
+	{
+		$filename = $this->getArgument(1);
+		$blocklist = $config->get('system', 'blocklist', []);
+		$fp = fopen($filename, 'w');
```
Comment (MrPetovan) [inline src/Console/ServerBlock.php:114]: Please check that the file was correctly open and output and error if it wasn't.

File: src/Console/ServerBlock.php (hunk 1 above)
Comment (tobiasd) [reply src/Console/ServerBlock.php:51]: IIRC because it was in a mixed state and spaces in the help block made more sense to me then tabs.

File: src/Console/ServerBlock.php [hunk 8]
```diff
@@ -87,17 +91,84 @@ protected function doExecute()
...
+				} else {
+					$reason = $data[1];
+				}
+				$data = [
+					'domain' => $domain,
+					'reason' => $reason
+				];
+				if (!in_array($data, $newBlockList)) {
+					$newBlockList[] = $data;
+                }
```
Comment (MrPetovan) [inline src/Console/ServerBlock.php:147]: Code standards: Wrong indentation, should be tabs.

File: src/Console/ServerBlock.php [hunk 9]
```diff
@@ -87,17 +91,84 @@ protected function doExecute()
...
+			}
+			foreach ($currBlockList as $blocked) {
+				if (!in_array($blocked, $newBlockList)) {
+					$newBlockList[] = $blocked;
+                }
```
Comment (MrPetovan) [inline src/Console/ServerBlock.php:152]: Code standards: Wrong indentation, should be tabs.
